Pass GH_PANERON_CI_TOKEN via env var

CAMOBAP opened the pull request:
Per https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

> If you must pass secrets within a command line, then enclose them within the proper quoting rules. Secrets often contain special characters that may unintentionally affect your shell. To escape these special characters, use quoting with your environment variables

CAMOBAP committed:
Pass GH_PANERON_CI_TOKEN via env var

CAMOBAP commented:
Something strange...

CAMOBAP commented:
I think it maybe somehow related to this https://github.com/paneron/paneron/blob/release/README.adoc#working-on-bundled-libraries

@strogonoff @ribose-jeffreylau maybe you can help explain this, do we still need it, or we should update package.json to remove "file:./dependencies-local/extension-kit-dist" and "file:./dependencies-local/registry-kit-dist" ?

ribose-jeffreylau commented:
@CAMOBAP Please rebase your commits from scratch (on the current `release`) again, since those lines you see should've been removed.

strogonoff commented:
@CAMOBAP the bug with dist libs was already fixed yesterday, check Zulip notifications :)

ribose-jeffreylau commented:
@CAMOBAP You may need to update your local `release` first before rebasing :)

CAMOBAP commented:
> @CAMOBAP the bug with dist libs was already fixed yesterday, check Zulip notifications :)

Maybe you can explain the commit https://github.com/paneron/paneron/pull/106/commits/a3db4f99f89efda15fb840fe02b7b17661ff7948 is it related to this PR?

ribose-jeffreylau commented:
@CAMOBAP That commit https://github.com/paneron/paneron/commit/a3db4f99f89efda15fb840fe02b7b17661ff7948 can be safely removed.

ribose-jeffreylau reviewed (commented):
I think to be able to test whether this token works, would be to add the trigger `on.push:` for this workflow, push so CI for this PR runs, and if it works, force push / revert the `on.push:` commit.  If it doesn't work, then continue working on this PR until it works, then finally force push / revert the `on.push:` commit, then merge into `release`.

CAMOBAP commented:
> I think to be able to test whether this token works, would be to add the trigger `on.push:` for this workflow, push so CI for this PR runs, and if it works, force push / revert the `on.push:` commit. If it doesn't work, then continue working on this PR until it works, then finally force push / revert the `on.push:` commit, then merge into `release`.

It works I suppose https://github.com/paneron/paneron/actions/runs/5877664665/job/15938203311

ribose-jeffreylau commented:
I see that it was run manually.  Since it directly references `$GH_PANERON_CI_TOKEN` then I guess it really works.

Cuz I think I've had experience when it was referencing `$GH_TOKEN` and would work when run manually, but would give `400` error when run via `workflow_call`.

CAMOBAP committed:
Add GH_PANERON_CI_TOKEN explicitly on workflow_call

CAMOBAP commented:
> I see that it was run manually. Since it directly references `$GH_PANERON_CI_TOKEN` then I guess it really works.
> 
> Cuz I think I've had experience when it was referencing `$GH_TOKEN` and would work when run manually, but would give `400` error when run via `workflow_call`.

@ribose-jeffreylau good catch, PR updated, let's test it on ubuntu, once passed I uncomment `if` condition

ribose-jeffreylau reviewed (commented):
https://github.com/paneron/paneron/actions/runs/5877984413/job/15939164062

I just tried manually running `update-websites.yml` on `release` branch and it succeeded, so I guess nothing more needs to be done here?  _I.e._, the updated token works well.

CAMOBAP commented:
> I just tried manually running update-websites.yml on release branch and it succeeded, so I guess nothing more needs to be done here? I.e., the updated token works well.

You are right `GH_PANERON_CI_TOKEN` works well 
But it's not accessible once workflow called with `workflow_call`, only with `workflow_dispatch` (as you mentioned above

I think instead explicit pass make sense to add `secrets: inherit` to `update-website` job as was done for `build-dist` in https://github.com/paneron/paneron/commit/4dc78a4a13785670bfaac9ba33f663cb09df966e

@ribose-jeffreylau make sense to you?

CAMOBAP commented:
Alson this one have to be adressed

<img width="1485" alt="Screenshot 2023-08-16 at 1 16 46 PM" src="https://github.com/paneron/paneron/assets/2169738/5da57f81-03d3-448a-97c5-750e359f5bae">

ribose-jeffreylau commented:
> > I just tried manually running update-websites.yml on release branch and it succeeded, so I guess nothing more needs to be done here? I.e., the updated token works well.
> 
> You are right `GH_PANERON_CI_TOKEN` works well But it's not accessible once workflow called with `workflow_call`, only with `workflow_dispatch` (as you mentioned above
> 
> I think instead explicit pass make sense to add `secrets: inherit` to `update-website` job as was done for `build-dist` in [4dc78a4](https://github.com/paneron/paneron/commit/4dc78a4a13785670bfaac9ba33f663cb09df966e)
> 
> @ribose-jeffreylau make sense to you?

You are right.  I agree that adding `secrets: inherit` should solve it.

ribose-jeffreylau commented:
> Alson this one have to be adressed
> <img alt="Screenshot 2023-08-16 at 1 16 46 PM" width="1485" src="https://user-images.githubusercontent.com/2169738/260990979-5da57f81-03d3-448a-97c5-750e359f5bae.png">

I added a concurrency group here thinking that only one `update-website` needs to succeed in the case of several such workflows running concurrently.

CAMOBAP committed:
Use 'secrets: inherit' to pass GH_PANERON_CI_TOKEN to update-website

CAMOBAP committed:
Remove concurrency group from update-website

It makese no sense because it can be executed as part of os-specific
workflow, and will inherit global concurrency group
Or it will be triggered manually

CAMOBAP commented:
> > Alson this one have to be adressed
> > <img alt="Screenshot 2023-08-16 at 1 16 46 PM" width="1485" src="https://user-images.githubusercontent.com/2169738/260990979-5da57f81-03d3-448a-97c5-750e359f5bae.png">
> 
> I added a concurrency group here thinking that only one `update-website` needs to succeed in the case of several such workflows running concurrently.

It will not help because probably they will not run at the same time anyway, ubuntu probably will finish faster

CAMOBAP committed:
Fix syntax error on ubuntu-latest, secrets defined twice

CAMOBAP commented:
Friendly reminder @strogonoff

strogonoff commented on .github/workflows/ubuntu-latest.yml:
```diff
@@ -66,3 +66,4 @@ jobs:
     # Update the website to show new version.
     if: ${{ startsWith(github.ref, 'refs/tags/v') }}
     uses: ./.github/workflows/update-website.yml
+    secrets: inherit
```
@CAMOBAP by the looks of it, your editor is not configured to not leave trailing newline? I think we generally use trailing newlines, it’s a good idea to stick to one approach to avoid unnecessary diffs I guess
